src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,242 @@
+                ["method"] = "alipay.user.info.share",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", new CultureInfo("zh-CN")),
+                ["version"] = "1.0",
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams, Options.ClientSecret));
+
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, sortedParams);
+
+            using var response = await Backchannel.GetAsync(address);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            using var payload = JsonDocument.Parse(await GetRawStringAsync(response, "alipay_user_info_share_response"));
+            string statusCode = payload.RootElement.GetString("code");
+
+            /* if code==10000 then success else failed. See: https://docs.open.alipay.com/common/105806 */
+            if (statusCode != "10000")
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                            "returned a {Status} response with the following message: {Message}.",
+                            /* Status: */ statusCode,
+                            /* Message: */ payload.RootElement.GetString("msg"));
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, payload.RootElement.GetString("user_id"), ClaimValueTypes.String, Options.ClaimsIssuer));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => string.Join(",", Options.Scope);
+
+        /// <summary>
+        /// Gets an HTTP payload as an asynchronous operation.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="elementName">Name of the element.</param>
+        private static async Task<string> GetRawStringAsync(
```

Comment:
Based on how you're using this (effectively double parsing some JSON), I think this would be better.

```csharp
private static async Task<JsonDocument> ReadJsonDocumentAsync(
    [NotNull] HttpResponseMessage message,
    [NotNull] string propertyName)
{
    using var document = JsonDocument.Parse(await message.Content.ReadAsStringAsync());
    string json = document.RootElement.GetProperty(propertyName).ToString();

    return JsonDocument.Parse(json);
}